### DISCUSSION TITLE
github will automatically delete my personal access token when I generate the .env file in ci

### DISCUSSION THREAD

**Discussion by qianxi0410** [community/community#30076] [Other Feature Feedback, Questions, & Ideas]
```yml
- name: Gen env file
  run: |
    touch .env.local
    echo DEPLOY_TARGET=gh-pages >> .env.local
    echo OWNER=${{ github.actor }} >> .env.local
    echo REPO=${{ github.event.repository.name }} >> .env.local
    echo ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }} >> .env.local
```

After running this ci step, my ACCESS_TOKEN will be automatically deleted by github.

**Comment by airtower-luna** [comment]
What happens to that file after you create it? Is there any chance it might get committed to a repository?

**Reply by qianxi0410** [reply]
will not be committed, only create-deploy-then delete the file. After this, my PAT is automatically deleted. That is, it will only be used once.

**Reply by qianxi0410** [reply]
Before deploying the step, I created this file, which is used during deployment, and immediately after deploying the step, I deleted the file.

**Reply by airtower-luna** [reply]
So the `.env` file is part of the deployment. What kind of deployment is it? What I suspect is that the GitHub secret scanning detects the PAT as leaked.
